src/Components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,277 @@
+
+  const changeView = (value: string) => {
+    if (value === "Month") {
+      setView(Views.MONTH);
+      setToggleValue("Month");
+    }
+    if (value === "Week") {
+      setView(Views.WEEK);
+      setToggleValue("Week");
+    }
+    if (value === "Day") {
+      setView(Views.DAY);
+      setToggleValue("Day");
+    }
+  };
+
+  const togglePopup = () => {
+    setShow(!show);
+  };
+
+  const eventClick = (event: any) => {
+    const newEvent = {
+      id: event.id,
+      title: event.title,
+      start: convertToDateTimeLocalString(event.start),
+      end: convertToDateTimeLocalString(event.end),
+    } as PopupEvent;
+    setEventPopup(newEvent);
+    togglePopup();
+  };
+
+  const handleSelectSlot = (event: any) => {
+    let endTime = event.end;
+    if (event.action === "click") {
+      endTime = moment(event.start).add(defaultEventLength, "m").toDate();
+    }
+    const newEvent = {
+      id: myEvents.slice(-1)[0].id + 1, //get next ID available after last element. Should be safe?
+      title: "",
+      start: convertToDateTimeLocalString(event.start),
+      end: convertToDateTimeLocalString(endTime),
+    } as PopupEvent;
+    setEventPopup(newEvent);
+    togglePopup();
+  };
+
+  const idExists = (array: CalendarEvent[], idToCheck: number): boolean => {
+    return array.some((obj) => obj.id === idToCheck);
+  };
+
+  const submitEvent = (event: CalendarEvent) => {
+    if (idExists(myEvents, event.id)) {
+      setEvents(myEvents.map((item) => (item.id === event.id ? event : item)));
+    } else {
+      setEvents([...myEvents, event]);
+    }
+    togglePopup();
+  };
+
+  const deleteEvent = (id: number) => {
+    setEvents(myEvents.filter((item) => item.id !== id));
+    togglePopup();
+  };
+
+  const components: any = {
+    event: ({ event }: any) => {
+      event.start = new Date(event.start);
+      event.end = new Date(event.end);
+      const title = event?.title;
+      const date = event?.start;
+      let past = false;
+      const currentDate = new Date();
+      if (currentDate > event?.end) {
+        past = true;
+      }
+      const timeString = `${date.getHours()}:${date.getMinutes().toString().padStart(2, "0")}`;
+      return (
+        <div className={`${styles.eventLabel} `}>
+          {view === "month" && (
+            <div className={styles.timeLabel}>{timeString}</div>
+          )}
+          <div className={styles.titleLabel}>{title}</div>
+        </div>
+      );
+    },
+    timeSlotWrapper: (timeSlotWrapperProps) => {
+      /*
+      //Future feature if needed
+      //Template for setting work hours highlight. 
+      
+      const hasCustomInfo = timeSlotWrapperProps.value
+        ? timeSlotWrapperProps.value.getHours() >= availability?.start &&
+          timeSlotWrapperProps.value.getHours() < availability?.end
+        : false;
+      */
+
+      const style = {
+        display: "flex",
+        flex: 1,
+        position: "relative",
+      };
```

Comment:
What is this used for?

Reply:
Originally for styling of timeSlots, currently unused.